### DISCUSSION TITLE
feature request: keybind to temporarily toggle visibility of tab bar and/or menu bar

### DISCUSSION THREAD

**Discussion by goproslowyo** [ghostty-org/ghostty#3907] [Feature Requests, Ideas] [answered]
In some ~~GTK~~ apps when you hide the menu bar, you can press `alt` to temporarily toggle the menu bar visibility.

In Ghostty, I have both the menu bar and tab bar hidden and it'd be nice if I could press `alt` to temporarily toggle the visibility of the menu bar and/or tab bar.

~~edit:
I thought `keybind = cmd+\=toggle_tab_overview` could be used as a workaround but it turns out that the `toggle_tab_overview` action doesn't even work if `gtk-titlebar = false`.~~ Resolved in a bugfix.

note that this differs from the toggle_tab_overview in that it shouldn't show the overview but the menu bar/tab bar instead.

a few examples of how other apps handle this:
> Google Chrome: Pressing the "Alt" key temporarily shows the menu bar in Chrome on Windows, but Chrome generally hides the menu bar by default, just like Firefox. On macOS, Chrome doesn't hide the menu bar in the same way because macOS always displays it at the top of the screen.

> Microsoft Office (Word, Excel, etc.): In recent versions of Office apps, the ribbon (which contains the menu options) is hidden by default in some modes. You can temporarily show the ribbon by pressing the "Alt" key or clicking on a tab. In addition, the ribbon will show up if you hover the mouse over it.

> Windows File Explorer: In some versions of Windows, the traditional menu bar is hidden by default. You can press "Alt" to show the menu bar temporarily. Once you click on a menu option or click elsewhere, the menu bar will disappear again.

> Mac Applications: Many macOS applications, like Safari or Finder, have a menu bar that is always visible at the top of the screen, but some allow the option to hide or auto-hide certain toolbars or controls. In certain contexts, pressing specific keys (like "Command" or a modifier key) can show or interact with hidden UI elements.

e.g. firefox
#### menubar hidden
![image](https://github.com/user-attachments/assets/49665144-c029-48b2-afc3-11ed85217273)

#### alt-pressed
![image](https://github.com/user-attachments/assets/681b8347-2639-4332-8869-6d1183dc0c12)

**Comment by pluiedev** [comment, marked as answer]
I think this is something GTK automatically does if you have a [PopoverMenuBar](https://docs.gtk.org/gtk4/class.PopoverMenuBar.html), but I might be wrong. See #2642 for more discussion about that.
